internal/data/database.go

```diff
@@ -0,0 +1,83 @@
+package data
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	_ "github.com/lib/pq"
+)
+
+type PostgresConfig struct {
+	Host         string
+	Port         string
+	User         string
+	Password     string
+	Database     string
+	SSLMode      string
+	MaxOpenConns int
+	MaxIdleConns int
+	MaxIdleTime  string
+}
+
+func (cfg PostgresConfig) String() string {
+	return fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=%s", cfg.Host, cfg.Port, cfg.User, cfg.Password, cfg.Database, cfg.SSLMode)
+}
+
+func OpenDB(cfg PostgresConfig) (*sql.DB, error) {
+	db, err := sql.Open("postgres", cfg.String())
+	if err != nil {
+		return nil, err
+	}
+
+	db.SetMaxOpenConns(cfg.MaxOpenConns)
+
+	db.SetMaxIdleConns(cfg.MaxIdleConns)
+
+	duration, err := time.ParseDuration(cfg.MaxIdleTime)
+	if err != nil {
+		return nil, err
+	}
+	db.SetConnMaxIdleTime(duration)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	err = db.PingContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return db, nil
+
+}
+
+func LoadDBConfig() (PostgresConfig, error) {
+	var cfg PostgresConfig
```

Comment:
Do you think functional options could be a fit here?

https://golang.cafe/blog/golang-functional-options-pattern.html